test-result-summary-client/src/Build/Summary/ResultSummary.jsx

```diff
@@ -53,16 +53,13 @@ export default class ResultSummary extends Component {
         });
 
         const builds = await fetchBuild.json();
-        const regex = /^Test_openjdk(\w+)_(\w+)_(\w+).(.+?)_(.+?_.+?(_xl)?)(_.+)?$/i;
         const buildMap = {};
         let jdkVersionOpts = [];
         let jdkImplOpts = [];
         builds.map((build, i) => {
             const buildName = build.buildName.toLowerCase();
-            const tokens = buildName.match(regex);
-            if (Array.isArray(tokens) && tokens.length > 5) {
-                const [_, jdkVersion, jdkImpl, level, group, platform] = tokens;
-
+            const {jdkVersion, jdkImpl, level, group, platform} = getInfoFromBuildName(buildName);
```

Comment:
Please check `getInfoFromBuildName` return value. This code will break if it returns null/